client/landing/stepper/declarative-flow/internals/steps-repository/goals/select-goals.tsx

```diff
@@ -0,0 +1,43 @@
+import { Onboard } from 'calypso/../packages/data-stores/src';
+import { useGoals } from './goals';
+import SelectCard from './select-card';
+
+type SelectGoalsProps = {
+	onChange: ( selectedGoals: Onboard.GoalKey[] ) => void;
+	selectedGoals: Onboard.GoalKey[];
+};
+
+export const SelectGoals: React.FC< SelectGoalsProps > = ( { onChange, selectedGoals } ) => {
+	const goals = useGoals();
+
+	const handleChange = ( selected: boolean, value: string ) => {
+		const newSelectedGoals = [ ...selectedGoals ];
+		const goalKey = value as Onboard.GoalKey;
+
+		if ( selected ) {
+			newSelectedGoals.push( goalKey );
+		} else {
+			const goalIndex = newSelectedGoals.indexOf( goalKey );
+			newSelectedGoals.splice( goalIndex, 1 );
+		}
+
+		onChange( newSelectedGoals );
+	};
+
+	return (
+		<div className="select-goals__container">
+			{ goals.map( ( goal ) => (
+				<SelectCard
+					onChange={ handleChange }
```

Comment:
I believe react's `key` prop is missing here on `SelectCard`, I could be mistaken?

Reply:
Good catch, just pushed a fix. Thank you!